### ISSUE TITLE
Should not be able to go to /dashboard without being logged in including authentication and authorization

### ISSUE THREAD

**Issue by gbowne1** [gbowne1/RadioLogger#41]
/dashboard should be protected route.

I am able to go directly to /dashboard manually without putting in my login by just adding /dashboard.

**Comment by jzunigarce** [comment]
This weekend I will implement the middleware to protect the routes.

**Comment by gbowne1** [comment]
great. I was looking up a bunch of stuff for that.

I'm working on making the reusable navbar component. in /src/client/public/coomponents/navbar/ right now.

**Comment by gbowne1** [comment]
I am thinking we might need something like this 

```js
app.post('/login', (req, res) => {
  // Perform authentication and authorization checks
  // If successful, redirect the user to the desired page
  if (authenticationSuccessful && authorizationSuccessful) {
    res.redirect('/dashboard');
  } else {
    // Handle authentication or authorization failure
    res.redirect('/login');
  }
});
```

**Comment by jzunigauabcs** [comment]
We can store the token in a cookie and validate it with a middleware on the server

**Comment by gbowne1** [comment]
yeah thats a good idea. @jzunigarce

**Comment by gbowne1** [comment]
I tried this.. it isn't really working as a redirect after /login authentication && authorization is good over to the route /dashboard.

**Comment by jzunigarce** [comment]
I added a middleware in the dashboard route and it doesn't let me enter directly without logging in, however I tried to create an account and log in on the front and it throws some errors, it won't let me log in.

**Comment by gbowne1** [comment]
yeah I had that same problem

**Comment by gbowne1** [comment]
I still register a user, then try to log in with that user and password, and click check button and then click login and it does not go to /dashboard automatically if the login is good.

**Comment by jzunigarce** [comment]
I hace a error on frontend. If login is successful and you use ajax you should redirect with js

**Comment by gbowne1** [comment]
I couldnt find a redirect anywhere.

**Comment by jzunigarce** [comment]
When you recibe a response of ok on login, you should redirect with window.location.replace

**Comment by gbowne1** [comment]
I was trying to avoid Ajax.  That sounds about right to me though.  Still feels to me like theres some bits missing.

**Comment by gbowne1** [comment]
res.redirect does not occur anywhere either even from a app.post or a router.post

**Comment by gbowne1** [comment]
I totally get the logged in object in the console... but theres no events to go with it

**Comment by jzunigarce** [comment]
The login in server was with the api, you should make with ajax, you can use fetch function. We hace SSR only for return page.

**Comment by gbowne1** [comment]
I am still getting a lot of wierd errors from the front end and every page in during the whole login process

**Comment by jzunigarce** [comment]
Can you take a ss of errors?

**Comment by gbowne1** [comment]
yeah

**Comment by gbowne1** [comment]
when on localhost:3000/login

![Screenshot_login_1](https://github.com/gbowne1/RadioLogger/assets/47549872/35e20166-16d3-49fe-858b-d86949667105)

**Comment by gbowne1** [comment]
when just on localhost:3000/
![Screenshot_main1](https://github.com/gbowne1/RadioLogger/assets/47549872/10ec66cf-bcd0-4610-93c5-7a085d627b48)

**Comment by jzunigarce** [comment]
The errors are from the js of frontend, we need refactory the js files

**Comment by gbowne1** [comment]
yeah, definitely.  Like I mentioned before the initial files were just from the original projects.  We still need to refactor around all the garbage/junk.

**Comment by gbowne1** [comment]
@jzunigarce I was able to fix  some of the errors but some still exist.  I added a app.use in the server.js to handle some issues with some .js files not getting the correct cors headers for xss, etc.  Login still needs a lot of work.

Lately I have not been able to go to the /dashboard route no matter I am logged in or out. I don't know what's up.

**Comment by gbowne1** [comment]
@ZennilGreenherald @shishir-21 @jzunigarce 

I haven't been able to verify that without putting in login details or having registered a user just going to /dashboard does or doesn't render.
